### DISCUSSION TITLE
Use with asyncio/trio

### DISCUSSION THREAD

**Discussion by Redoubts** [Textualize/rich#1056] [Q&A]
Naive question, but are there any sharp edges I might encounter if I use `rich` in something using asyncio/trio? I don't see it mentioned much, but I suspect there would be a few issues in the tracker if there was something big to worry about.

**Comment by willmcgugan** [comment]
Haven't used Rich with asyncio, but I don't anticipate you'll have any major problems.

Writing to a file, even stdout, is actually a blocking operation. In practice though the OS buffers things so it won't block unless you are writing massive amounts of text to the terminal. There are a few file operations in Rich. If you use `save_html` or `save_text` to write console contents to a file, that will also block. It's unlikely to be a bottleneck, but you have the option of running those in a thread.
